pkg/schema/definitions/visitor_test.go

```diff
@@ -0,0 +1,245 @@
+			wantField: definitionField{
+				Type:        protoPrimitive.Type,
+				Description: protoPrimitive.Description,
+			},
+		},
+		{
+			name:            "integer primitive",
+			inputSchema:     &protoPrimitiveInt,
+			wantDefinitions: map[string]definition{},
+			wantField: definitionField{
+				Type:        "int",
+				Description: protoPrimitiveInt.Description,
+			},
+		},
+		{
+			name:            "number primitive",
+			inputSchema:     &protoPrimitiveNumber,
+			wantDefinitions: map[string]definition{},
+			wantField: definitionField{
+				Type:        "int",
+				Description: protoPrimitiveNumber.Description,
+			},
+		},
+		{
+			name:        "kind",
+			inputSchema: &protoKind,
+			wantDefinitions: map[string]definition{
+				protoKind.Path.String(): {
+					ResourceFields: map[string]definitionField{
+						"protoArray": {
+							Type:        "array",
+							Description: protoArray.Description,
+							SubType:     protoPrimitive.Type,
+							Required:    true,
+						},
+						"protoMap": {
+							Type:        "map",
+							Description: protoMap.Description,
+							SubType:     protoPrimitive.Type,
+						},
+						"protoPrimitive": {
+							Type:        protoPrimitive.Type,
+							Description: protoPrimitive.Description,
+							Required:    true,
+						},
+						"protoRef": {
+							Type:        protoKind.Path.String(),
+							Description: protoRef.Description,
+						},
+					},
+					Type:        protoKind.Path.String(),
+					Description: protoKind.Description,
+				},
+			},
+			wantField: definitionField{
+				Description: protoKind.Description,
+				Type:        protoKind.Path.String(),
+			},
+		},
+		{
+			name:            "reference no subschema",
+			inputSchema:     &protoRefNoSubSchema,
+			wantDefinitions: map[string]definition{},
+			wantField: definitionField{
+				Type:        protoRefNoSubSchema.reference,
+				Description: protoRefNoSubSchema.Description,
+			},
+		},
+		{
+			name:        "reference",
+			inputSchema: &protoRef,
+			wantDefinitions: map[string]definition{
+				protoKind.Path.String(): {
+					ResourceFields: map[string]definitionField{
+						"protoArray": {
+							Type:        "array",
+							Description: protoArray.Description,
+							SubType:     protoPrimitive.Type,
+							Required:    true,
+						},
+						"protoMap": {
+							Type:        "map",
+							Description: protoMap.Description,
+							SubType:     protoPrimitive.Type,
+						},
+						"protoPrimitive": {
+							Type:        protoPrimitive.Type,
+							Description: protoPrimitive.Description,
+							Required:    true,
+						},
+						"protoRef": {
+							Type:        protoKind.Path.String(),
+							Description: protoRef.Description,
+						},
+					},
+					Type:        protoKind.Path.String(),
+					Description: protoKind.Description,
+				},
+			},
+			wantField: definitionField{
+				Type:        protoKind.Path.String(),
+				Description: protoRef.Description,
+			},
+		},
```

Comment:
If that makes sense, you may want to add cases for some nested types as well.

Reply:
I'm going to avoid doing this for now since we've agreed that we are going to keep the current behavior, which we would like to potentially change in the future. If you disagree and would like to see a test case regardless, let me know.